src/models/slots_iterator/iterator.rs

```diff
@@ -16,12 +20,29 @@ impl Iterator for TimeSlotsIterator {
                     get_start_of_repeat_step(&self.current_start_position, repetition);
                 let mut indexes_to_delete_count: usize = 0;
                 for mut slot in self.timeline.slots.clone().into_iter() {
+                    dbg!(&next_start_position, &slot);
                     if next_start_position.le(&slot.end) && next_start_position.gt(&slot.start) {
-                        //next_start_position is 'on' the current slot
-                        result.slots.insert(Slot {
-                            start: slot.start,
-                            end: next_start_position,
-                        });
+                        // TODO 2023-07-07: Use TimingScenario to avoid wrongly slots for overflow slots
+                        if let Some(filters) = &self.filters {
+                            // Determine the timing scenario based on the `after_time` and `before_time` inputs
+                            let timing_scenario =
+                                determine_timing_scenario(filters.after_time, filters.before_time);
```

Comment:
The iterator should not have to know about `timing_scenario` (after_time or before_time).  

This sleep-1 case is more about adjusting the step generation.  
Concretely, just add an adjustment at line 54 of `step_generator`:  
- subtract a few hours (difference between after and 0:00) from `start`
- add a few hours (different between 0:00 and before) to `deadline`